hal/src/common/thumbv7em/clock.rs

```diff
@@ -205,9 +205,9 @@ impl GenericClockController {
             state.reset_gclk();
             state.set_gclk_divider_and_source(GCLK1, 1, XOSC32K, false);
         } else {
-            enable_internal_32kosc(osc32kctrl);
-            state.reset_gclk();
-            state.set_gclk_divider_and_source(GCLK1, 1, OSCULP32K, false);
+            //enable_internal_32kosc(osc32kctrl);
```

Comment:
Hmmm?

Reply:
On the SAMD5x, GCLK1 is the only generic clock generator whose maximum division factor is 2^16. All the others are 2^9.  See [14.8.3](http://ww1.microchip.com/downloads/en/DeviceDoc/60001507E.pdf#unique_5042). If DFLL is used as the input to GCLK1, then we can have a clock frequency 48MHz/[1..2^16]. 48kHz happens to be a standard audio sampling rate, so dividing the 48MHz clock by 1000 (only possible with GCLK1) would achieve an exact 48kHz ADC clock with minimal fuss. Unfortunately this doesn't account for the clock division that the ADC performs, but in certain cases it might be possible to achieve a 48kHz sample rate. Since GCLK1 is hardcoded by the HAL as the 32kHz oscillator, I disabled it to investigate this clocking approach.

Reply:
Ah okies, just remember to uncomment when youre ready!